### ISSUE TITLE
Provide openapi 3.0 spec for API

### ISSUE THREAD

**Issue by giautm** [pocketbase/pocketbase#208]
I want to access the APIs via Golang/PHP. If we can provide the specification via OpenAPI spec. I can generate the client using existing tools. You also generate the server code using OpenAPI tools.
For example: https://github.com/ogen-go/ogen

**Comment by ganigeorgiev** [comment]
Although I can see the benefit of OpenApi (in the past I've even written a php generator for the v2 spec), in reality the end result may not be very practical and it could also sometimes obfuscate the code if you keep the documentation as comments alongside the handlers.

I don't think generating an OpenApi docs for PocketBase will be very useful in general because developers most likely are going to use mainly the [Records](https://pocketbase.io/docs/api-records/) and the [Realtime](https://pocketbase.io/docs/api-realtime/) services and they are fairly small and easy to wrap. Additionally, since the collection fields are user defined, we'll have to keep the generated OpenApi schema somehow in sync during runtime (as a hook with cache?), which may complicate further the internals.

In short - I'm against adding support for OpenApi for PocketBase since it will be hard to maintain and keep in sync with the existing more human readable documentation and examples (which will be expanded even further in the near future).

I'll convert the issue to discussion and if someone disagree and would like to see OpenApi supported, please feel free to share your feedback (it really helps me to understand how developers prefer to work with PocketBase).
